examples/step-85/step-85.cc

```diff
@@ -0,0 +1,548 @@
+      // iterate over the local faces, we first check if the current face
+      // belongs to the set $\mathcal{F}_h$. The actual assembly is simple using
+      // FEInterfaceValues. Assembling in this we will traverse each internal
+      // face in the mesh twice, so in order to get the penalty constant we
+      // expect, we multiply the penalty term with a factor 1/2.
+      for (unsigned int f : cell->face_indices())
+        if (face_has_ghost_penalty(cell, f))
+          {
+            const double face_side_length =
+              cell->face(f)->minimum_vertex_distance();
+
+            const unsigned int invalid_subface = numbers::invalid_unsigned_int;
+
+            fe_interface_values.reinit(cell,
+                                       f,
+                                       invalid_subface,
+                                       cell->neighbor(f),
+                                       cell->neighbor_of_neighbor(f),
+                                       invalid_subface);
+
+            const unsigned int n_interface_dofs =
+              fe_interface_values.n_current_interface_dofs();
+            FullMatrix<double> local_stabilization(n_interface_dofs,
+                                                   n_interface_dofs);
+            for (unsigned int q = 0;
+                 q < fe_interface_values.n_quadrature_points;
+                 ++q)
+              {
+                const Tensor<1, dim> normal = fe_interface_values.normal(q);
+                for (unsigned int i = 0; i < n_interface_dofs; ++i)
+                  for (unsigned int j = 0; j < n_interface_dofs; ++j)
+                    {
+                      local_stabilization(i, j) +=
+                        .5 * ghost_parameter * face_side_length * normal *
+                        fe_interface_values.jump_in_shape_gradients(i, q) *
+                        normal *
+                        fe_interface_values.jump_in_shape_gradients(j, q) *
+                        fe_interface_values.JxW(q);
+                    }
+              }
+
+            const std::vector<types::global_dof_index>
+              local_interface_dof_indices =
+                fe_interface_values.get_interface_dof_indices();
+
+            stiffness_matrix.add(local_interface_dof_indices,
+                                 local_stabilization);
+          }
+    }
+}
+
+
+// @sect3{Solving the System}
+template <int dim>
+void Step85<dim>::solve()
+{
+  const unsigned int max_iterations = solution.size();
+  SolverControl      solver_control(max_iterations);
+  SolverCG<>         solver(solver_control);
+  solver.solve(stiffness_matrix, solution, rhs, PreconditionIdentity());
+}
+
+
+
+// @sect3{Data Output}
+// Since both DoFHandler instances use the same triangulation, we can add both
+// the level set function and the solution to the same vtu-file.
+template <int dim>
+void Step85<dim>::output_results() const
+{
+  DataOut<dim> data_out;
```

Comment:
Maybe only output the relevant cells? You could use `DataOut::set_cell_selection()`.